### PR TITLE
Refactor STS

### PR DESCRIPTION
**Pull request by fatalistt**
1. Fix typos
2. Remove extra configuration interfaces and service provider
3. Throw exceptions rather logging errors while setting Signing Credentials, cause without that we will get a non-working app (it'll throw an exception that there is no signing credentials)
4. Throw exceptions rather logging errors while setting Validation Key, cause we get a bad configured app if we will continue.
5. Refactor startup and program.cs to be more ".net core 3"-way.

### CODE REVIEW

File: src/Skoruba.IdentityServer4.STS.Identity/Program.cs [hunk 1]
```diff
@@ -8,14 +9,21 @@ public class Program
...
+                .ConfigureLogging((hostingContext, logging) => {
+                    logging.ClearProviders();
+                    var logger = new LoggerConfiguration().ReadFrom.Configuration(hostingContext.Configuration).CreateLogger();
+                    logging.AddSerilog(logger);
+                })
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.ConfigureKestrel(options => options.AddServerHeader = false);
+                    webBuilder.UseStartup<Startup>();
+                    webBuilder.UseSerilog();
```
Comment (duki994) [inline src/Skoruba.IdentityServer4.STS.Identity/Program.cs]: @fatalistt 

You don't need _UseSerilog_ on _IWebHostBuilder_

Comment (fatalistt) [reply src/Skoruba.IdentityServer4.STS.Identity/Program.cs]: Sorry for a long anwer, I was busy at work.
Yes, you're right, `IWebHostBuilder.UseSerilog` is redunant.

Comment (skoruba) [reply src/Skoruba.IdentityServer4.STS.Identity/Program.cs]: Can you remove this line before final merge, please? :-)
Thanks

Comment (fatalistt) [reply src/Skoruba.IdentityServer4.STS.Identity/Program.cs]: Already done that :)

**Comment by skoruba** [comment]
Great work, thanks!

**Pull request merged**